clients/v1_repl/v1_repl.go

```diff
@@ -0,0 +1,912 @@
+        format <format>       specifies the format of the server responses: json, csv, or column
+        precision <format>    specifies the format of the timestamp: h, m, s, ms, u or ns
+        history               displays command history
+        settings              outputs the current settings for the shell
+        clear                 clears settings such as database or retention policy.  run 'clear' for help
+        exit/quit/ctrl+d      quits the influx shell
+
+        show databases        show database names
+        show series           show series information
+        show measurements     show measurement information
+        show tag keys         show tag key information
+        show field keys       show field key information
+
+        A full list of influxql commands can be found at:
+        https://docs.influxdata.com/influxdb/latest/query_language/spec/`)
+	// ? Maybe add keybindings information like <CTRL+L> clearing the screen, <CTRL+D> exitting, etc
+}
+
+func (c *Client) settings() {
+	w := new(tabwriter.Writer)
+	w.Init(os.Stdout, 0, 1, 1, ' ', 0)
+	fmt.Fprintln(w, "Setting\tValue")
+	fmt.Fprintln(w, "--------\t--------")
+	// TODO: No supported way to access or change host & port in a REPL session
+	// if c.Port > 0 {
+	// 	fmt.Fprintf(w, "Host\t%s:%d\n", c.Host, c.Port)
+	// } else {
+	// 	fmt.Fprintf(w, "Host\t%s\n", c.Host)
+	// }
+	fmt.Fprintf(w, "Username\t%s\n", c.Username)
+	fmt.Fprintf(w, "Database\t%s\n", c.Database)
+	fmt.Fprintf(w, "RetentionPolicy\t%s\n", c.RetentionPolicy)
+	fmt.Fprintf(w, "Pretty\t%v\n", c.Pretty)
+	fmt.Fprintf(w, "Format\t%s\n", c.Format)
+	fmt.Fprintf(w, "Precision\t%s\n", c.Precision)
+	fmt.Fprintln(w)
+	w.Flush()
+}
+
+func (c *Client) query(ctx context.Context, query string) (string, error) {
+	var resContentType string
+	switch c.Format {
+	case CsvFormat:
+		resContentType = "application/csv"
+	case JsonFormat, TableFormat:
+		resContentType = "application/json"
+	default:
+		return "", fmt.Errorf("unexpected format: %s", c.Format)
+	}
+	resBody, err := c.GetLegacyQuery(ctx).
+		U(c.Username).
+		P(c.Password).
+		Db(c.Database).
+		Q(query).
+		Rp(c.RetentionPolicy).
+		Accept(resContentType).
+		Execute()
+	if err != nil {
+		return "", err
+	}
+	return resBody, nil
+}
+
+func (c *Client) setFormat(args []string) {
+	// args[0] is "format"
+	if len(args) != 2 {
+		color.Red("Expected a format [csv, json, table]")
+		return
+	}
+	newFormat := FormatType(args[1])
+	switch newFormat {
+	case CsvFormat, JsonFormat, TableFormat:
+		c.Format = newFormat
+	default:
+		color.HiRed("Unimplemented format %q, keeping %s format.", newFormat, c.Format)
+		color.HiBlack("Choose a format from [csv, json, table]")
+	}
+}
+
+func (c *Client) setPrecision(args []string) {
+	// args[0] is "precision"
+	if len(args) != 2 {
+		color.Red("Expected a precision [ns, u, ms, s, m, or h]")
+		return
+	}
+	precision := args[1]
+	switch precision {
+	case "ns", "u", "µ", "ms", "s", "m", "h":
```

Comment:
Are we allowing hour/minute precision? I thought I saw above that we were not but that could have been an unrelated section of code

Reply:
Where did you see that we weren't allowing h/m precision?

Reply:
```
ERR: 400 Bad Request: invalid precision; valid precision units are ns, us, ms, and s
```